stdlib/test/collections/test_string.mojo

```diff
@@ -1253,15 +1253,6 @@ def test_string_iter():
         concat += v
     assert_equal(321, atol(concat))
 
-    for v in vs:
```

Comment:
**Question** Why remove this test?

Reply:
It was actually a test that shouldn't be possible. The compiler won't let you do this now
```mojo
for v in vs: # immutable ref iterator
    v.unsafe_ptr()[] = ord("1") # this is  mutating the origin
```